src/Serializer/AbstractItemNormalizer.php

```diff
@@ -249,7 +249,34 @@ public function denormalize($data, $class, $format = null, array $context = [])
             return $item;
         }
 
-        return parent::denormalize($data, $resourceClass, $format, $context);
+        $objectToPopulate = $context[self::OBJECT_TO_POPULATE] ?? null;
+        $isUpdate = null !== $objectToPopulate;
+
+        // Filter out input attributes that aren't allowed for updates
+        if ($isUpdate) {
+            $data = array_filter($data, function (string $attribute) use ($context, $objectToPopulate) {
```

Comment:
Why filtering the input? Was it possible to modify a property of a resource without the right to do so before?

Reply:
You're right that there was already a security pass - `AbstractObjectNormalizer::denormalize()` calls `AbstractItemNormalizer::getAllowedAttributes()` but only with the class string, so `object` can't be used. This addition does another pass over, calling the `security` expression with a populated `object` variable. Without this, `security` will always be called with a null object for updates, which may be unexpected. We could take this out and force use of `security_post_denormalize` if the developer wants an object, but I think an object would be expected for updates as it already exists.

Reply:
I'd be in favor of reducing the complexity in our code, enforcing the use of `security_post_denormalize`